src/ipc/ipc4/helper.c

```diff
@@ -374,6 +370,19 @@ int ipc_comp_connect(struct ipc *ipc, ipc_pipe_comp_connect *_connect)
 		return IPC4_FAILURE;
 	}
 
+	/* create a buffer
+	 * in case of LL -> LL or LL->DP
+	 *	size = 2*obs of source module (obs is single buffer size)
+	 * in case of DP -> LL
+	 *	size = 2*ibs of destination (LL) module. DP queue will handle obs of DP module
+	 */
+	uint32_t buf_size;
+
+	if (source->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_LL)
+		buf_size = source_src_cfg.obs * 2;
+	else
+		buf_size = sink_src_cfg.ibs * 2;
```

Comment:
does ibs and obs already 2 times of actual need? if so, do we still need *2 here?

Reply:
IBS and OBS are strict values that come from the IPC call
I supposed to be **exact** value that is 
 - needed for module to process 1 cycle (IBS)
 - amount of data produced by the module in 1 cycle (OBS)

2* comes from this https://github.com/thesofproject/sof/pull/5047

Reply:
Hmm, I don't get the link to #5047 ? So some entities don't understand difference between valid and container sizes, and for that we double memory size for all existing LL->LL usages?
Please tell me I'm missing something here.

Reply:
I simply did not want to make any changes to way that LL pipeline is working in this PR.
Double buffer size LL was introduced as a bugfix and is already in current code:

```
static struct comp_buffer *ipc4_create_buffer(struct comp_dev *src, bool is_shared,
					      uint32_t src_obs, uint32_t src_queue,
					      uint32_t dst_queue)
{
	struct sof_ipc_buffer ipc_buf;
	int buf_size;

	/* double it since obs is single buffer size */
	buf_size = src_obs * 2;
```

After my changes LL it is working in the very same way. Regardless if it is right or not - I really don't think changing it in this PR is a good idea. It may be in separate PR

For DP - I have just followed the same pattern, maybe a little too fast. I'll double check if 2* it brings any value here